homeassistant/components/vicare/climate.py

```diff
@@ -210,6 +208,7 @@ def update(self) -> None:
 
             with suppress(PyViCareNotSupportedFeatureError):
                 self._attributes["vicare_modes"] = self._circuit.getModes()
+            self._attributes["vicare_programs"] = self._circuit.getPrograms()
```

Comment:
Why do we update the `vicare_programs` attribute but not the `_attr_preset_modes`?
If the programs can't change, we should probably set both in `__init__`, if they can change, we should change both here?

Also, can't call to `getPrograms` raise `PyViCareNotSupportedFeatureError`?

Reply:
> Also, can't the call to getPrograms raise PyViCareNotSupportedFeatureError?

`getPrograms` iterates over a fixed list of known programs, if no `PyViCareNotSupportedFeatureError` is raised, the program is considered as supported and added to result list.

Reply:
I see no reason why this should change, so will move it to `init`.